### ISSUE TITLE
Refactor calls to `bmi_model->GetValue<T>()` to `bmi::model::GetValue<T>()`

### ISSUE THREAD

**Issue by mattw-nws** [NOAA-OWP/ngen#463]
Refactor to eliminate deprecation.

## Current behavior

Numerous deprecation warnings in ngen and test code builds similar to the following:

```
/ngen/ngen/src/realizations/catchment/Bmi_Fortran_Formulation.cpp:57:73: warning: 
'std::vector<T> models::bmi::Bmi_Fortran_Adapter::GetValue(const string&) [with T = long double; std::__cxx11::string = std::__cxx11::basic_string<char>]' is deprecated: 
Functionality moved to models::bmi::GetValues in bmi_utilities.hpp [-Wdeprecated-declarations]
```
(NOTE: There is a typo in the deprecation message, there is no `models::bmi::GetValues` _plural_, it should be `models::bmi::GetValue` _singular_ ... but the object is to elminate the messages completely.)

the deprecated class method is called in several places in ngen and test code. 

## Expected behavior

Migrate all calls to use `bmi::model::GetValue<T>()` instead. See [Bmi_Module_Formulation.hpp](https://github.com/NOAA-OWP/ngen/commit/3fed52379abd55eafa30d48138a232663cad400e#diff-ae1264775c477c87bced0f5ed3e382966fe721ec49ee36e7498e86fd01b9727dR413) for an example.

Eliminate all `GetValue` deprecation messages during build

Finally, Remove deprecated `GetValue` functions (may also be able to remove any functions exclusively used by them, e.g. `inner_get_value`).

## Steps to replicate behavior (include URLs)

1.

**Comment by mattw-nws** [comment]
There's a chance this migration may exposes issues... especially in the Fortran code, maybe. We will have to watch test results and check the final result somewhat carefully.

**Comment by stcui007** [comment]
Looks like this will need testing of Fortran code which will involve more
data files, especially namelist.input.

**Comment by mattw-nws** [comment]
If you are running, for example, a version of Noah-owp-modular that still isn't pinned in master, those files still need to stay out of commits. We probably need to update the pin, but that's a separate PR.

**Comment by stcui007** [comment]
I assume by
bmi_model->GetValue

maybe you meant (more precisely)
get_bmi_model->GetValue?

**Comment by mattw-nws** [comment]
>maybe you meant (more precisely)
>get_bmi_model->GetValue?

No, that would be less precise... where something like that occurs it's `get_bmi_model()->GetValue...` (a method call not a variable pointer) but also, don't solve this with find and replace. I can't be much more precise as it happens because there are actually three separate methods with the same name but no common ancestor. However all three have `[[deprecated(...)]]` messages on them and have the same signature.

**Comment by mattw-nws** [comment]
Closed in #467
